edge-23.10.2 change notes

alpeb opened the pull request:
This edge release includes a fix addressing an issue during upgrades for instances not relying on automated webhook certificate management (like cert-manager provides).

* Added a `checksum/config` annotation to the destination and proxy injector deployment manifests to force restarting those workloads whenever their webhook secrets change during upgrade (thanks @iAnomaly!) ([#11440])
* Fixed policy controller error when deleting a gateway API HTTPRoute resource ([#11471])

hawkw commented on CHANGES.md:
```diff
@@ -1,5 +1,20 @@
+cert-manager provides).
+
+* Added a `checksum/config` annotation to the destination and proxy injector
+  deployment manifests to force restarting those workloads whenever their
```
```suggestion
  deployment manifests, to force restarting those workloads whenever their
```

hawkw commented on CHANGES.md:
```diff
@@ -1,5 +1,20 @@
+* Added a `checksum/config` annotation to the destination and proxy injector
+  deployment manifests to force restarting those workloads whenever their
+  webhook secrets change during upgrade (thanks @iAnomaly!) ([#11440])
+* Fixed policy controller error when deleting a gateway API HTTPRoute resource
```
```suggestion
* Fixed policy controller error when deleting a Gateway API HTTPRoute resource
```

hawkw reviewed (approved):
modulo some tiny changelog edits, this looks good to me!

alpeb commented:
## edge-23.10.2

This edge release includes a fix addressing an issue during upgrades for
instances not relying on automated webhook certificate management (like
cert-manager provides).

* Added a `checksum/config` annotation to the destination and proxy injector
  deployment manifests, to force restarting those workloads whenever their
  webhook secrets change during upgrade (thanks @iAnomaly!) ([#11440])
* Fixed policy controller error when deleting a Gateway API HTTPRoute resource
  ([#11471])

[#11440]: https://github.com/linkerd/linkerd2/pull/11440
[#11471]: https://github.com/linkerd/linkerd2/pull/11471
